lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -370,7 +370,14 @@ function _campaign_resource_reportback($nid, $values) {
     return $rb_info['rbid'];
   }
   else {
+    // Increment transaction id and log that the request has been received with new transaction id.
+    $transaction_id_parts = explode('-', $headers['X-Request-Id']);
```

Comment:
Maybe add a check that this header exists before working with it? Or that `transaction_id_parts` is not an empty array before trying to work with it?